x-pack/plugins/observability_solution/inventory/server/routes/entities/get_entity_types.ts

```diff
@@ -16,14 +15,21 @@ export async function getEntityTypes({
 }: {
   inventoryEsClient: ObservabilityElasticsearchClient;
 }) {
-  const entityTypesEsqlResponse = await inventoryEsClient.esql<{
-    entity: Pick<EntityInstance['entity'], 'type'>;
-  }>('get_entity_types', {
-    query: `FROM ${ENTITIES_LATEST_ALIAS}
+  const entityTypesEsqlResponse = await inventoryEsClient.esql<
+    {
+      'entity.type': string;
+    },
+    { transform: 'plain' }
```

Comment:
just noticed we need to pass the `transform` here and in the arguments. Is there a way to simplify this?

Reply:
I'm not aware of any. TS doesn't support partial inference (AFAIK, see https://github.com/microsoft/TypeScript/issues/26242).